chainbase/src/main/java/org/tron/core/db2/core/SnapshotImpl.java

```diff
@@ -48,21 +54,36 @@ public byte[] get(byte[] key) {
   }
 
   private byte[] get(Snapshot head, byte[] key) {
+    Histogram.Timer requestTimer = Metrics.histogramStartTimer(
+            MetricKeys.Histogram.SNAPSHOT_SERVICE_LATENCY, root.getDbName(), MetricLabels.Histogram.SNAPSHOT_GET);
     Snapshot snapshot = head;
     Value value;
     if (isOptimized) {
       value = db.get(Key.of(key));
+      if(Objects.nonNull(value)){
+        Metrics.histogramObserve(requestTimer);
+      }
+      Metrics.counterInc(MetricKeys.Counter.SNAPSHOT_GET,1,root.getDbName(),
+              Objects.isNull(value) ? MetricLabels.Counter.SNAPSHOT_GET_MISS : MetricLabels.Counter.SNAPSHOT_GET_SUCCESS,
+              MetricLabels.Counter.SNAPSHOT_GET_NOT_REACH_ROOT);
       return value == null ? null: value.getBytes();
     }
     while (Snapshot.isImpl(snapshot)) {
       if ((value = ((SnapshotImpl) snapshot).db.get(Key.of(key))) != null) {
+        Metrics.histogramObserve(requestTimer);
+        Metrics.counterInc(MetricKeys.Counter.SNAPSHOT_GET,1,root.getDbName(),MetricLabels.Counter.SNAPSHOT_GET_SUCCESS,
+                MetricLabels.Counter.SNAPSHOT_GET_NOT_REACH_ROOT);
         return value.getBytes();
       }
 
       snapshot = snapshot.getPrevious();
     }
 
-    return snapshot == null ? null : snapshot.get(key);
+    byte[] result = snapshot == null ? null : snapshot.get(key);
+    Metrics.counterInc(MetricKeys.Counter.SNAPSHOT_GET,1,root.getDbName(),
+            Objects.isNull(result)?MetricLabels.Counter.SNAPSHOT_GET_MISS:MetricLabels.Counter.SNAPSHOT_GET_SUCCESS,
+            MetricLabels.Counter.SNAPSHOT_GET_REACH_ROOT);
```

Comment:
same as MetricKeys.Counter.DB_GET ?

Reply:
> same as MetricKeys.Counter.DB_GET ?

SNAPSHOT_GET is used to measure the snapshotImpl performance while the DB_GET used to measure the snapshotRoot performance.